xapian-core/weight/bm25weight.cc

```diff
@@ -71,8 +83,13 @@ BM25Weight::init(double factor)
 	double numerator = (reltermfreq + 0.5) * (Q - tf + 0.5);
 	double denom = (reldocs_not_indexed + 0.5) * (nonreldocs_indexed + 0.5);
 	tw = numerator / denom;
-    } else {
-	tw = (get_collection_size() - tf + 0.5) / (tf + 0.5);
+	}
+	else if(bm25+) {
```

Comment:
`else if` should be on the same line as the closing brace, and there should be a space after `if`. Also, the alignment is wrong on your new lines (which is why this is difficult to read in github as effectively adding a few lines).